synapse/storage/util/id_generators.py

```diff
@@ -447,6 +521,28 @@ def _add_persisted_position(self, new_id: int):
                 # do.
                 break
 
+    def _update_stream_positions_table_txn(self, txn):
+        """Update the `stream_positions` table with newly persisted position.
+        """
+
+        if not self._writers:
+            return
```

Comment:
Doesn't really matter, but this is checked in the callers too. I'm assuming this was just double checking?

Reply:
Yes, just in case somewhere forgets to do the check.